Added pure canny as another mode

RaghavC1 opened the pull request:
Added another mode that just spits back out the camera-processing output. I was going to make it just black/white, but I have left it as-is for now since I thought it looked cooler. 

Let me know if it needs to be re-factored in any way, I didn't put much thought into it.

RaghavC1 commented:
Added a simple web UI that can always be easily altered and doesn't need to interact with the python code. Can be expanded to control the actual exhibit computer in any way.

chucky1729 commented:
Hey Raghav,

I certainly like the simplicity of adding this debug mode via a "simulation". Great idea!

I have a few suggestions:
1. To me, the blue flickering display is a sign that something's not working as expected. I was wondering if it was just a Mac thing but it sounds like you see the same thing on Windows. I'd bet that the problem is that processed image is grayscale (i.e. one dimensional pixels) but pygame is expecting RGB (i.e. three dimensional pixels). Did you try broadcasting the single grayscale value to an RGB triplet value?
2. Instead of rotating and mirroring the image to convert from numpy's row major order to pygame's column major order (which is f'ing weird, by the way), I think you should just transpose the frame. It's probably slightly faster, and it's more "correct" in my opinion.
3. Is there a better name for the file than canny.py? Like image.py or passthrough.py? Currently we're doing more than just canny edge detection and in the future we might not even be using canny if it turns out that one of the other edge detection algorithms has better properties for our use case.
4. Try to make the coding convention match that of what's already been written. What stood out to me was the spacing around assignments and the use of spaces instead of tabs.

That said, these are all just suggestions! I'm also okay with you just merging your changes as is.

Cheers,
Charlie

RaghavC1 commented:
Hey Charlie, 

1. I actually didn't try to fix the blue flickering issue earlier because I thought it looked cool :), but yeah the issue is that opencv gives pixels in BGR instead of RGB which pygame expects, so I went ahead and fixed that.
2. I have no idea why I didn't just think of checking if opencv already had a way of transposing an image already instead of doing it myself. I went ahead and did it that way, I agree it's probably faster and more "correct."
3. You're right, I changed the name of the mode to passthrough, feels better.
4. AH I thought I had fixed the tabs/spaces issues earlier. Also fixed spaces around assignments.

Gonna go ahead and merge
